drivers/volume/kdmp/kdmp.go

```diff
@@ -231,26 +239,174 @@ func (k *kdmp) UpdateMigratedPersistentVolumeSpec(
+			Kind:       reflect.TypeOf(storkapi.BackupLocation{}).Name(),
+			Name:       restore.Spec.BackupLocation,
+			Namespace:  restore.Namespace,
+			APIVersion: StorkAPIVersion,
+		}
+		volBackup.Spec.Repository = fmt.Sprintf("%s/%s-%s/", prefixRepo, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		volBackup.Status.SnapshotID = bkpvInfo.BackupID
+		if _, err := kdmpShedOps.Instance().CreateVolumeBackup(volBackup); err != nil {
+			logrus.Errorf("unable to create backup cr: %v", err)
+			return nil, err
+		}
+
+		// create kdmp cr
+		dataExport := &kdmpapi.DataExport{}
+		dataExport.Annotations = make(map[string]string)
+		dataExport.Annotations[skipResourceAnnotation] = "true"
+		dataExport.Name = getGenericCRName(restore.Name, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		dataExport.Namespace = bkpvInfo.Namespace
+		dataExport.Status.TransferID = volBackup.Namespace + "/" + volBackup.Name
+		dataExport.Spec.Type = kdmpapi.DataExportKopia
+		dataExport.Spec.Source = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(kdmpapi.VolumeBackup{}).Name(),
+			Name:       volBackup.Name,
+			Namespace:  volBackup.Namespace,
+			APIVersion: KdmpAPIVersion,
+		}
+		dataExport.Spec.Destination = kdmpapi.DataExportObjectReference{
+			Kind:       PVCKind,
+			Name:       bkpvInfo.PersistentVolumeClaim,
+			Namespace:  bkpvInfo.Namespace,
+			APIVersion: "v1",
+		}
+		if _, err := kdmpShedOps.Instance().CreateDataExport(dataExport); err != nil {
+			logrus.Errorf("failed to create DataExport CR: %v", err)
+			return volumeInfos, err
+		}
+		volumeInfos = append(volumeInfos, volumeInfo)
+	}
 	return volumeInfos, nil
 }
 
-func (k *kdmp) CancelRestore(*storkapi.ApplicationRestore) error {
+func (k *kdmp) CancelRestore(restore *storkapi.ApplicationRestore) error {
+	for _, vInfo := range restore.Status.Volumes {
+		crName := getGenericCRName(restore.Name, vInfo.SourceNamespace, vInfo.PersistentVolumeClaim)
+		if err := kdmpShedOps.Instance().DeleteDataExport(crName, vInfo.SourceNamespace); err != nil {
+			logrus.Errorf("failed to delete data export CR: %v", err)
+			return err
+		}
+	}
 	return nil
 }
 
 func (k *kdmp) GetRestoreStatus(restore *storkapi.ApplicationRestore) ([]*storkapi.ApplicationRestoreVolumeInfo, error) {
 	volumeInfos := make([]*storkapi.ApplicationRestoreVolumeInfo, 0)
+	for _, vInfo := range restore.Status.Volumes {
+		if vInfo.DriverName != driverName {
+			continue
+		}
+		crName := getGenericCRName(restore.Name, vInfo.SourceNamespace, vInfo.PersistentVolumeClaim)
+		dataExport, err := kdmpShedOps.Instance().GetDataExport(crName, vInfo.SourceNamespace)
+		if err != nil {
+			logrus.Errorf("failed to get DataExport CR: %v", err)
+			return volumeInfos, err
+		}
+		if dataExport.Status.TransferID == "" {
+			vInfo.Status = storkapi.ApplicationRestoreStatusInitial
+			vInfo.Reason = "Volume restore not started yet"
+		} else {
+			if isBackupActive(dataExport.Status) {
```

Comment:
suggestion:It is good to change the function names isBackupActive and isBackupComplete as it is used in both backup and restore. Possible isDataExportActive and isDataExportCompleted?

Reply:
yes, updated